### PR TITLE
25 setup screen

### PR DESCRIPTION
**Pull request by lchristus**

### CODE REVIEW

File: src/MensaWidget/MWDockingBarTest.class.st [hunk 1]
```diff
@@ -43,6 +43,22 @@ MWDockingBarTest >> tearDown [
...
+{
+	#category : #tests,
+	#'squeak_changestamp' : 'LC 7/5/2024 00:56'
+}
+MWDockingBarTest >> testOpenSettings [
+
+	| menu submenu widget |
+	
+	TheWorldMainDockingBar showMensaMenuInWorldMainDockingBar: true.
+	self dockingbar allMorphsDo: [:each | (each name asString includesSubstring: 'MensaMenu') ifTrue: [menu := each]].
```
Comment (tommccaughey) [inline src/MensaWidget/MWDockingBarTest.class.st:55]: line indent

**Review by tommccaughey — CHANGES_REQUESTED** [review]

File: src/MensaWidget/MWDockingBarTest.class.st (hunk 1 above)
Comment (Kangaroo6942) [reply src/MensaWidget/MWDockingBarTest.class.st:55]: why exactly would this be indented?

**Review by Kangaroo6942 — CHANGES_REQUESTED** [review]
Thanks for implementing this! Didn't spot any bigger flaws, @lchristus and I already talked about a change in the `updatePrice`. Another thing I noticed is that the PopUp changes colour if you open another window in the meantime, any clue on how to stop that from happening?
In addition I noticed that in case of using the popUp to change the priceGroup with an open MensaWidget, the MenuBar-entry for the priceGroup doesn't change like it does when changing it there directly.

File: src/MensaWidget/MWDockingBarTest.class.st (hunk 1 above)
Comment (lchristus) [reply src/MensaWidget/MWDockingBarTest.class.st:55]: its tbh too long for one line

Comment (Kangaroo6942) [reply src/MensaWidget/MWDockingBarTest.class.st:55]: ah, i thought Tom just wanted to indent the self dockingbar line

File: src/MensaWidget/MWConfiguration.class.st [hunk 2]
```diff
@@ -7,10 +7,101 @@ Instance Variables
...
+	^ popUp
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'LC 7/3/2024 00:22'
+}
+MWConfiguration class >> isInitial [
+
+	^ IsInitial ifNil: [IsInitial := true]
```
[the remark this answers is not in the document]
Comment (lchristus) [reply src/MensaWidget/MWConfiguration.class.st:93]: this is exactly styled like the color vars with default values so I would keep it like that

[the remark this answers is not in the document]
Comment (Kangaroo6942) [reply src/MensaWidget/MWConfiguration.class.st:93]: which means we not only want to return a default value ifNil, but also set the value to the Default which is kind of redundant?

File: src/MensaWidget/MWDockingBarTest.class.st [hunk 3]
```diff
@@ -43,6 +43,22 @@ MWDockingBarTest >> tearDown [
...
 
+{
+	#category : #tests,
+	#'squeak_changestamp' : 'LC 7/5/2024 00:56'
+}
+MWDockingBarTest >> testOpenSettings [
+
+	| menu submenu widget |
+	
+	TheWorldMainDockingBar showMensaMenuInWorldMainDockingBar: true.
```
[the remark this answers is not in the document]
Comment (lchristus) [reply src/MensaWidget/MWDockingBarTest.class.st:70]: I agree its done

File: src/MensaWidget/MWDockingBarTest.class.st (hunk 1 above)
Comment (lchristus) [reply src/MensaWidget/MWDockingBarTest.class.st]: I wrote a helper function for that functionality i think that makes the code easier to read

File: src/MensaWidget/MWConfiguration.class.st (hunk 2 above)
[the remark this answers is not in the document]
Comment (lchristus) [reply src/MensaWidget/MWConfiguration.class.st:93]: i mean no from that point on the variable has a value which I personally find quite helpful

**Comment by lchristus** [comment]
I have also pushed a solution to the changing category issue, would be nice to also test that functionality

**Comment by Kangaroo6942** [comment]
i think there is a problem with setting the 'settings' content to an empty string when the settings actually are still nil

**Comment by lchristus** [comment]
> settings

i think those are the garbage collector ones because they wouldnt have settings set so settings is nil. Also settings doesnt refer to the setting but the setting popUp morph so the value of the actual setting doesnt matter. I can put an ifNotNil but its just on your machine with the garbage collector afaik

**Review by Kangaroo6942 — APPROVED** [review]
All resolved. Thanks for implementing the nil handling, I will talk to the teaching chair, why the garbage collector doesn't do shit

**Review by OscarUlb — APPROVED** [review]
Thanks for your work! Looks good!

**Review by tommccaughey — APPROVED** [review]

**Review by NilsKasper — APPROVED** [review]

**Pull request merged**